html5lib/html5parser.py

```diff
@@ -1706,6 +1706,7 @@ def startTagColgroup(self, token):
 
         def startTagCol(self, token):
             self.startTagColgroup(impliedTagToken("colgroup", "StartTag"))
+            token["selfClosingAcknowledged"] = True
```

Comment:
So some things return tokens to reprocess them in another state, such as for `col` in the "in table" insertion mode. Seemingly, none of the `token["selfClosingAcknowledged"]` assignments we have currently happen anywhere where a token is returned.

However, given the spec only defines the `col` start tag token as having it acknowledged in only one place (the "in column group" insertion mode), we should ideally do likewise, and the reprocessing of tokens in the `mainLoop` should handle this. This means we somehow need to keep track of the previous value of `new_token` given as you point out it's always `None` after the loop (you can tell I didn't look closely!), presumably by adding a `prev_token` variable or some such.